### PR TITLE
refactor: migrate rpc::common to use trait RpcMethod

### PR DESCRIPTION
**Pull request by aatifsyed**
## Summary of changes

Changes introduced in this pull request:

-

## Reference issue to close (if applicable)

Closes

## Change checklist

- [ ] I have performed a self-review of my own code,
- [ ] I have made corresponding changes to the documentation,
- [ ] I have added tests that prove my fix is effective or that my feature works (if possible),
- [ ] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

**Review by ruseinov — APPROVED** [review]

File: src/cli/main.rs [hunk 1]
```diff
@@ -45,7 +45,7 @@ where
                 Subcommand::Info(cmd) => cmd.run(api).await,
                 Subcommand::Snapshot(cmd) => cmd.run(api).await,
                 Subcommand::Attach(cmd) => cmd.run(api),
-                Subcommand::Shutdown(cmd) => cmd.run(api).await,
+                Subcommand::Shutdown(cmd) => cmd.run(rpc::Client::from(api)).await,
```
[the remark this answers is not in the document]
Comment (aatifsyed) [reply src/cli/main.rs:48]: I think this is clearer :)

**Review by LesnyRumcajs — APPROVED** [review]

File: src/tool/subcommands/api_cmd.rs [hunk 2]
```diff
@@ -971,25 +971,28 @@ async fn start_offline_server(
...
+    rpc_port: u16,
+    mut shutdown_recv: mpsc::Receiver<()>,
+) -> anyhow::Result<()>
 where
     DB: Blockstore + Send + Sync + 'static,
 {
     info!("Starting offline RPC Server");
     let rpc_address = SocketAddr::new(IpAddr::V4(Ipv4Addr::UNSPECIFIED), rpc_port);
-    let forest_version = FOREST_VERSION_STRING.as_str();
-    let (shutdown_send, mut shutdown_recv) = mpsc::channel(1);
```
Comment (lemmih) [inline src/tool/subcommands/api_cmd.rs:988]: This fixes a bug, right? Was the offline RPC server correctly shutdown before?

**Review by lemmih — APPROVED** [review]
+5

**Review by ruseinov — APPROVED** [review]

File: src/tool/subcommands/api_cmd.rs (hunk 2 above)
Comment (aatifsyed) [reply src/tool/subcommands/api_cmd.rs:988]: I don't think there was a bug here - I just needed to reorg `RPCState` to fit the new way of doing things :)

**Pull request merged**